src/data-connect/data-connect-api-client-internal.ts

```diff
@@ -0,0 +1,233 @@
+  ): Promise<ExecuteGraphqlResponse<GraphqlResponse>> {
+    return this.executeGraphqlHelper(query, EXECUTE_GRAPH_QL_READ_ENDPOINT, options);
+  }
+
+  private async executeGraphqlHelper<GraphqlResponse, Variables>(
+    query: string,
+    endpoint: string,
+    options?: GraphqlOptions<Variables>,
+  ): Promise<ExecuteGraphqlResponse<GraphqlResponse>> {
+    if (!validator.isNonEmptyString(query)) {
+      throw new FirebaseDataConnectError(
+        'invalid-argument',
+        '`query` must be a non-empty string.');
+    }
+    if (typeof options !== 'undefined') {
+      if (!validator.isNonNullObject(options)) {
+        throw new FirebaseDataConnectError(
+          'invalid-argument', 'GraphqlOptions must be a non-null object');
+      }
+    }
+    const host = (process.env.DATA_CONNECT_EMULATOR_HOST || DATA_CONNECT_HOST);
+    const data = {
+      query,
+      ...(options?.variables && { variables: options?.variables }),
+    };
+    return this.getUrl(host, this.connectorConfig.location, this.connectorConfig.serviceId, endpoint)
+      .then(async (url) => {
+        const request: HttpRequestConfig = {
+          method: 'POST',
+          url,
+          headers: DATA_CONNECT_CONFIG_HEADERS,
+          data,
+        };
+        const resp = await this.httpClient.send(request);
+        return Promise.resolve({
+          data: resp.data.data as GraphqlResponse,
+        });
+      })
+      .then((resp) => {
+        return resp;
+      })
+      .catch((err) => {
+        throw this.toFirebaseError(err);
+      });
+  }
+
+  private async getUrl(host: string, locationId: string, serviceId: string, endpointId: string): Promise<string> {
+    return this.getProjectId()
+      .then((projectId) => {
+        const urlParams = {
+          host,
+          projectId,
+          locationId,
+          serviceId,
+          endpointId
+        };
+        const baseUrl = utils.formatString(DATA_CONNECT_API_URL_FORMAT, urlParams);
+        return utils.formatString(baseUrl);
+      });
+  }
+
+  private getProjectId(): Promise<string> {
+    if (this.projectId) {
+      return Promise.resolve(this.projectId);
+    }
+    return utils.findProjectId(this.app)
+      .then((projectId) => {
+        if (!validator.isNonEmptyString(projectId)) {
+          throw new FirebaseDataConnectError(
+            'unknown-error',
+            'Failed to determine project ID. Initialize the '
+            + 'SDK with service account credentials or set project ID as an app option. '
+            + 'Alternatively, set the GOOGLE_CLOUD_PROJECT environment variable.');
+        }
+        this.projectId = projectId;
+        return projectId;
+      });
+  }
+
+  private toFirebaseError(err: RequestResponseError): PrefixedFirebaseError {
+    if (err instanceof PrefixedFirebaseError) {
+      return err;
+    }
+
+    const response = err.response;
+    if (!response.isJson()) {
+      return new FirebaseDataConnectError(
+        'unknown-error',
```

Comment:
Why not use the map here? `DATA_CONNECT_ERROR_CODE_MAPPING.UNKNOWN`?

Reply:
Thanks! Fixed